Orange/widgets/visualize/widgetutils/tree/rules.py

```diff
@@ -0,0 +1,181 @@
+class Rule:
+    """The base Rule class for tree rules."""
+
+    def merge_with(self, rule):
+        """Merge the current rule with the given rule.
+
+        Parameters
+        ----------
+        rule : Rule
+
+        Returns
+        -------
+        Rule
+
+        """
+        raise NotImplemented()
+
+
+class DiscreteRule(Rule):
+    """Discrete rule class for handling Indicator rules.
+
+    Parameters
+    ----------
+    attr_name : str
+    eq : bool
+        Should indicate whether or not the rule equals the value or not.
+    value : object
+
+    Examples
+    --------
+    Age = 30
+    >>> rule = DiscreteRule('age', True, 30)
+    Name ≠ John
+    >>> rule = DiscreteRule('name', False, 'John')
```

Comment:
The order, yes. For such examples, it is customary to simulate the python interpreter session:
```py
>>> something I type in
something the interpreter spews out
```
For example:
```py
>>> rule = DiscreteRule('name', False, 'John')
>>> print(rule)
Age = 30
```
(Could also be without `print()` if `rule` prints that on `repr()` call.)